tap_adroll/client.py

```diff
@@ -12,27 +12,39 @@
 
 class AdrollAuthenticationError(Exception):
     pass
-
-
 class AdrollClient():
-    def __init__(self, config_path, config):
-        token = {
-            'access_token': config['access_token'],
-            'refresh_token': config['refresh_token'],
-            'token_type': 'Bearer',
-            # Set expires_in to a negative number to force the client to reauthenticate
-            'expires_in': '-30'
-        }
-        extra = {
-            'client_id': config['client_id'],
-            'client_secret': config['client_secret']
-        }
+    def __init__(self, config_path, config, dev_mode = False):
+        self.dev_mode = dev_mode
         self.config_path = config_path
-        self.session = OAuth2Session(config['client_id'],
-                                     token=token,
-                                     auto_refresh_url=TOKEN_REFRESH_URL,
-                                     auto_refresh_kwargs=extra,
-                                     token_updater=self._write_config)
+        self.config = config
+        self.authenticate_request()
+
+    def authenticate_request(self):
+        if self.dev_mode :
+            self.session = requests.Session()
+            read_config = self._read_config()
```

Comment:
Why do we have to read the config **_again_**?

`authenticate_request()` is only called from `AdrollClient.__init__()`

The `config` gets parsed in `tap_adroll.__init__.py` and is passed to the `AdrollClient` constructor. So you can just pull the `"access_token"` off of `self.config`

Reply:
Agreed, made the changes to get `access_token` from `self.config` and hence, removed the `_read_config()`